mlflow/sklearn/__init__.py

```diff
@@ -1544,8 +1544,8 @@ def patched_fit(fit_impl, original, self, *args, **kwargs):
             and _AUTOLOGGING_METRICS_MANAGER.should_log_post_training_metrics()
         )
 
-        with _SklearnTrainingSession(clazz=self.__class__, allow_children=False) as t:
-            if t.should_log():
+        with _SklearnTrainingSession(estimator=self, allow_children=False) as t:
+            if t.should_log() and not t.is_current_estimator_nested_call():
```

Comment:
The extra condition `not t.is_current_estimator_nested_call()` is for avoiding autologging twice. Still uses the example: suppose we calling `lgbm_classifier_instance.fit`

```
(1) patched LGBMClassifier.fit ->  (t.should_log() == True)
(2) original LGBMClassifier.fit -> 
(3) patched LGBMModel.fit ->  (t.should_log() == True)  then autologging twice
(4) original LGBMModel.fit ->
```

Reply:
@WeichenXu123 are there any cases where we *do* want to log when `t.is_current_estimator_nested_call()` is true? If not, can we update the `should_log` implementation to:

```
            for training_session in _TrainingSession._session_stack:
                if training_session is self:
                    break
                elif training_session.estimator is self.estimator:
                    return False

            return self._parent is None or self._parent.allow_children
```

Reply:
@dbczumar 
> are there any cases where we do want to log when t.is_current_estimator_nested_call() is true? 

Yes. The below (5) place. Code see here: https://github.com/mlflow/mlflow/pull/7404/files#r1038722373

(1) patched LGBMClassifier.fit ->  We want enable autologging patch here.
(2) original LGBMClassifier.fit -> 
(3) patched LGBMModel.fit ->  We don't want enable autologging patch here.
(4) original LGBMModel.fit ->
(5) patched lightgbm.train ->  We want log here.
(5) original lightgbm.train

Reply:
Shouldn't this be solved by implementing the recommendation in my last comment and then setting allow_children=True when creating the TrainingSession object in the patch for LGBMClassifier?

Reply:
@dbczumar Updated according to your suggestion. :)